src/com/walmartlabs/lacinia/pedestal.clj

```diff
@@ -299,45 +334,102 @@
                 (assoc context :response (bad-request (message-as-errors "Subscription queries must be processed by the WebSockets endpoint.")))
                 context))}))
 
+(defn ^:private pass-query-through-query-store
+  [compiled-schema query-store query-cache]
+  (let [*cache (atom query-cache)]
+    (fn [context]
+      (cond-let
+        :let [{q :graphql-query
+               operation-name :graphql-operation-name} (:request context)
+              ;; Often, the q here is a full query document, resulting in a cache miss.
+              ;; Only when q identifies a valid query document will the *parsed* query be
+              ;; present in the cache.
+              cache-key [q operation-name]
+              cached (cache/lookup (swap! *cache
+                                          #(if (cache/has? % cache-key)
+                                             (cache/hit % cache-key)
+                                             (cache/miss % cache-key nil)))
+                                   cache-key)]
+
+        (some? cached)
+        (assoc-in context parsed-query-key-path cached)
+
+        :let [store-result (query-store q)]
+
+        ;; Not a query name, an actual query; continue as normal.
+        (nil? store-result)
+        context
+
+        :else
+        (go
```

Comment:
We cannot use async interceptor as this makes the entire chain run on core.async threads. This has to be a regular sync interceptor instead.

Reply:
I've been thinking about that, and what we need to do is is set the async processing flag.  But I'm nervous about that too, because we're missing a piece of the puzzle related to handling uncaught exceptions.

Reply:
Though exceptions inside Lacinia code itself is rare, and applications are supposed to ensure that exceptions do not occur inside resolvers.

Reply:
What do you mean by "set the async processing flag"? Also, I don't really see the benefit of this being async in the first place.